include/circt/Dialect/HW/HWStructure.td

```diff
@@ -254,7 +254,8 @@ def HWModuleExternOp : HWModuleOpBase<"module.extern"> {
                        OptionalAttr<DictArrayAttr>:$res_attrs,
                        StrArrayAttr:$argNames, StrArrayAttr:$resultNames,
                        ParamDeclArrayAttr:$parameters,
-                       OptionalAttr<StrAttr>:$verilogName);
+                       OptionalAttr<StrAttr>:$verilogName,
+                       OptionalAttr<DictionaryAttr>:$esi_wrap);
```

Comment:
Drive by comment: why does anything ESI-specific need to be in the `hw` dialect? If you really need to attach something dialect-specific to an hw extern module, please use a dialect attribute.

Reply:
So this would mean to add a dialect attribute for example called esiWrap. Should I add this attribute to the dialect ESI or HW?

Reply:
Sorry, I meant discardable attribute. They're called "dialect attributes" in the API and "discardable attributes" in the documentation. It means they're not listed in the op definition. (And can be dropped by passes.)